libsolidity/lsp/LanguageServer.cpp

```diff
@@ -186,6 +186,16 @@ void LanguageServer::changeConfiguration(Json::Value const& _settings)
 			lspAssert(false, ErrorCode::InvalidParams, "Invalid file load strategy: " + text);
 	}
 
+	if (auto traceLogFileNode = _settings["trace-log-file"])
+	{
+		if (traceLogFileNode.isString())
+			m_client.setTraceLogFile(boost::filesystem::path(traceLogFileNode.asString()));
+		else if (traceLogFileNode.isNull())
+			m_client.setTraceLogFile(nullopt);
```

Comment:
Is this necessary? I.e. can `nullopt` simply be a default value for `m_traceLogFilePath`?

Reply:
I guess it's needed in case the client sends `changeConfiguration()` multiple times. If we set the default, it will only be `nullopt` the first time.